packages/edit-site/src/components/actions/view-actions.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { external } from '@wordpress/icons';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { ACTION_TYPES } from './constants';
+
+export const viewPostAction = {
+	id: 'view-post',
+	label: __( 'View' ),
+	type: ACTION_TYPES.primary,
+	icon: external,
+	isEligible( post ) {
+		return post.status !== 'trash';
```

Comment:
I checked how this is currently managed in wp-admin and only `trash` posts don't have a preview (private or scheduled ones do, for example). It sounds good as it is.